docs/codeql/codeql-overview/codeql-changelog/codeql-cli-2.16.4.rst

```diff
@@ -0,0 +1,156 @@
+
+Java
+""""
+
+*   To reduce the number of false positives in the query "Insertion of sensitive information into log files" (:code:`java/sensitive-log`), variables with names that contain "null" (case-insensitively) are no longer considered sources of sensitive information.
+
+Ruby
+""""
+
+*   Calls to :code:`Object#method`, :code:`Object#public_method` and :code:`Object#singleton_method` with untrusted data are now recognised as sinks for code injection.
+*   Added additional request sources for Ruby on Rails.
+
+New Queries
+~~~~~~~~~~~
+
+Java
+""""
+
+*   Added a new query :code:`java/android/insecure-local-key-gen` for finding instances of keys generated for biometric authentication in an insecure way.
+
+Python
+""""""
+
+*   The query :code:`py/nosql-injection` for finding NoSQL injection vulnerabilities is now part of the default security suite.
+
+Language Libraries
+------------------
+
+Bug Fixes
+~~~~~~~~~
+
+Golang
+""""""
+
+*   Fixed dataflow out of a :code:`map` using a :code:`range` statement.
+
+Java
+""""
+
+*   Fixed the Java autobuilder overriding the version of Maven used by a project when the Maven wrapper :code:`mvnw` is in use and the :code:`maven-wrapper.jar` file is not present in the repository.
+*   Some flow steps related to :code:`android.text.Editable.toString` that were accidentally disabled have been re-enabled.
+
+Swift
+"""""
+
+*   Fixed an issue where :code:`TypeDecl.getFullName` would get stuck in an loop and fail when minor database inconsistencies are present.
+
+Major Analysis Improvements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+C#
+""
+
+*   Improved support for flow through captured variables that properly adheres to inter-procedural control flow.
+*   We no longer make use of CodeQL database stats, which may affect join-orders in custom queries. It is therefore recommended to test performance of custom queries after upgrading to this version.
+
+Golang
+""""""
+
+*   We have significantly improved the Go autobuilder to understand a greater range of project layouts, which allows Go source files to be analysed that could previously not be processed.
+*   Go 1.22 has been included in the range of supported Go versions.
```

Comment:
When debugging earlier today with @angelapwen I wondered whether we should have a similar entry for C# (C# 12 and .NET 8 support are now documented). We do mention C# 12 in the detailed changelog items below.

Reply:
Is C# 12 new with this release?

Reply:
I think we're good to proceed without. Let's establish an internal guidance for these items soon, then we can be more strict about it. I'll mention it in the blog post of course.